plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -542,7 +542,7 @@ private void testCorruptedTableLocation(String tableName, Path tableLocation, bo
 
         // Assert queries fail cleanly
         assertQueryFails("TABLE " + tableName, "Metadata not found in transaction log for tpch." + tableName);
-        assertQueryFails("SELECT * FROM \"" + tableName + "$history\"", ".* Table '.*\\$history' does not exist");
+        assertThat(query("SELECT * FROM \"" + tableName + "$history\"")).returnsEmptyResult();
```

Comment:
Why it no longer fails?

Reply:
This is a side effect of not fetching the metadata in `DeltaLakeMetadata#getSystemTable` for history tables.

The expectations of what to do for corrupted tables is not super clear to me but empty results instead of an error for querying system tables didn't seem like a problematic behavior change to me.

Reply:
does it mean we produce no rows even if transaction log not found at all?
sounds like too much short-circuiting. i guess easy to address